oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/util/Utils.java

```diff
@@ -274,6 +277,61 @@ private static String diagsForEntry(Map.Entry<String, PropertyStats> member) {
         }
     }
 
+    /**
+     * Produce an {@link UpdateOp} suitable for shrinking branch revision entries for given property in {@link Document}, {@code null} otherwise.
+     * 
+     * @param doc document to inspect for repeated branch commits
+     * @param propertName property to check for
+     * @param revisionChecker filter for revisions (for instance, to check for cluster id)
+     * @return {@link UpdateOp} suitable for shrinking document, {@code null} otherwise
+     */
+    public static @Nullable UpdateOp getShrinkOp(Document doc, String propertyName, Predicate<Revision> revisionChecker) {
```

Comment:
If `doc` could be a `NodeDocument` instead of the generic `Document` some of those `instanceof` could be avoided...

Reply:
I *think* we can do that once we use that from NodeDocumentStore, not DocumentStore...